python/foglamp/tasks/north/sending_process.py

```diff
@@ -374,884 +284,654 @@ def __init__(self, loop=None):
+                .payload()
+            await self._storage_async.insert_into_tbl("statistics", payload)
+            payload = payload_builder.PayloadBuilder() \
+                .SELECT("key", "description") \
+                .WHERE(['key', '=', key]) \
+                .LIMIT(1) \
+                .payload()
+            statistics = await self._storage_async.query_tbl_with_payload("statistics", payload)
+            rows = statistics['rows']
+            return rows[0]['key']
 
-    async def _update_statistics(self, num_sent, stream_id):
-        """ Updates FogLAMP statistics
-        Raises :
-        """
         try:
-            key = 'SENT_' + str(stream_id)
-            _stats = await statistics.create_statistics(self._storage_async)
-
-            await _stats.update(key, num_sent)
+            payload = payload_builder.PayloadBuilder() \
+                .SELECT("key", "description") \
+                .WHERE(['key', '=', self._name]) \
+                .LIMIT(1) \
+                .payload()
+            statistics = await self._storage_async.query_tbl_with_payload("statistics", payload)
+            rows = statistics['rows']
+            statistics_key = await add_statistics(self._name, self._name) if len(rows) == 0 else rows[0]['key']
+        except Exception as e:
+            SendingProcess._logger.error(_MESSAGES_LIST["e000013"].format(str(e)))
+            raise e
+        return statistics_key
 
+    def _is_north_valid(self):
+        """ Checks if the north has adequate characteristics to be used for sending of the data"""
+        north_ok = False
+        try:
+            if self._plugin_info['type'] == self._PLUGIN_TYPE and \
+                            self._plugin_info['name'] != "Empty North Plugin":
+                north_ok = True
         except Exception:
-            _message = _MESSAGES_LIST["e000010"]
-            SendingProcess._logger.error(_message)
+            SendingProcess._logger.error(_MESSAGES_LIST["e000000"])
             raise
-
-    @staticmethod
-    def performance_track(message):
-        """ Tracks information for performance measurement
-        Args:
-        Returns:
-        Raises:
-        """
-
-        if _log_performance:
-            usage = resource.getrusage(resource.RUSAGE_SELF)
-            process_memory = usage.ru_maxrss / 1000
-
-            SendingProcess._logger.debug("PERFORMANCE - {0} : memory MB |{1:>8,}|".format(
-                                                                message,
-                                                                process_memory))
+        return north_ok
 
     def _plugin_load(self):
-        """ Loads the plugin
-        Args:
-        Returns:
-        Raises:
-        """
-        module_to_import = self._NORTH_PATH + self._config['north'] + "." + self._config['north']
+        module_to_import = self._NORTH_PATH + self._config['plugin'] + "." + self._config['plugin']
```

Comment:
it would be great to use .format style for better readablity

Reply:
done